clicommand/util_keygen.go

```diff
@@ -0,0 +1,111 @@
+package clicommand
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/buildkite/agent/v3/internal/jwkutil"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/urfave/cli"
+	"golang.org/x/exp/slices"
+)
+
+type KeygenConfig struct {
+	Alg                  string `cli:"alg" validate:"required"`
+	KeyID                string `cli:"key-id" validate:"required"`
+	PrivateKeyFilename   string `cli:"private-key-filename" normalize:"filepath"`
+	PublicKeysetFilename string `cli:"public-keyset-filename" normalize:"filepath"`
+
+	LogLevel string `cli:"log-level"`
+	Debug    bool   `cli:"debug"`
+}
+
+var KeygenCommand = cli.Command{
+	Name:  "keygen",
+	Usage: "Generate a new key pair, used to sign and verify jobs",
```

Comment:
Maybe a link to some docs on what the key and keyset formats are would be useful.

Reply:
ooohhhhh yep, definitely need to add a usage string to this, that information can go there.